NEW Specify If the Object Manages Lines for Module Builder

atm-quentin opened the pull request:
**Summary**
#32460 
This update introduces a configuration option to specify whether an object manages lines (e.g., invoices). If the object does not manage lines, related code generation is skipped.

atm-quentin committed:
Specify If the Object Manages Lines for Module Builder

eldy committed:
Merge branch 'develop' into FIX/32460/HandleLines

eldy committed:
Merge branch 'develop' into FIX/32460/HandleLines

eldy committed:
Merge branch 'develop' into FIX/32460/HandleLines

eldy committed:
Merge branch 'develop' into FIX/32460/HandleLines

eldy commented on htdocs/modulebuilder/index.php:
```diff
@@ -2052,7 +2075,7 @@ function getLicenceHeader($user, $langs, $now)
 			'scripts/myobject.php' => 'scripts/'.strtolower($objectname).'.php',
 			'class/myobject.class.php' => 'class/'.strtolower($objectname).'.class.php',
-			'class/api_myobject.class.php' => 'class/api_'.strtolower($module).'.class.php',
+			'class/api_myobject.class.php' => strtolower($module),
```
Why changing this line ?

eldy replied:
The PR generates warning in CI. But it is not possible to fix them because your PR was not created with permission to others to contributes.

atm-quentin replied:
it's an error sorry

atm-quentin replied:
```suggestion
			'class/api_myobject.class.php' => 'class/api_'.strtolower($module).'.class.php',
```

atm-quentin replied:
> The PR generates warning in CI. But it is not possible to fix them because your PR was not created with permission to others to contributes.

I enabled "Allow edits and access to secrets by maintainers"

eldy replied:
I confirm pr is open for fixing now, it will sabe time on both side, thanks....

atm-quentin committed:
Update htdocs/modulebuilder/index.php

eldy committed:
Merge branch 'develop' into FIX/32460/HandleLines

atm-quentin commented:
Are you waiting for an action from me? If so, which one?

eldy committed:
Merge branch 'develop' into FIX/32460/HandleLines

eldy committed:
Update myobject.class.php

eldy committed:
Merge branch 'develop' into FIX/32460/HandleLines

eldy committed:
Update myobject.class.php

eldy committed:
Update myobject.class.php

eldy committed:
Update myobject.class.php

eldy committed:
Update myobject.class.php

eldy committed:
Merge branch 'develop' into FIX/32460/HandleLines

eldy committed:
Update myobject.class.php

eldy committed:
Update myobject.class.php

eldy committed:
Update myobject.class.php

eldy committed:
Merge branch 'develop' into FIX/32460/HandleLines

eldy committed:
Merge branch 'develop' into FIX/32460/HandleLines

atm-quentin committed:
Merge branch 'develop' of https://github.com/Dolibarr/dolibarr into FIX/32460/HandleLines

eldy committed:
Merge branch 'develop' into FIX/32460/HandleLines

atm-quentin committed:
Fix modulebuilder CI issues

atm-quentin committed:
Merge remote-tracking branch 'origin/develop' into FIX/32460/HandleLines

# Conflicts:
#	htdocs/modulebuilder/index.php

atm-quentin committed:
Fix Phan PhanUndeclaredProperty errors on PR #32664

- pdf_standard_myobject: add @phan-suppress-current-line on situation_percent,
  localtax2_tx, localtax1_type, localtax2_type, vat_src_code, pagebreak
  (MyObjectLine template class does not declare these inherited properties)
- doc_generic_project_odt: replace fk_soc with socid (correct PHP property;
  fk_soc is the DB column name, Project::fetch() maps it to $this->socid)
